Sources/TestsCommon/CommonTestUtils.swift

```diff
@@ -338,7 +333,7 @@ public func cleanEqual(_ expectedValue: String?) -> Predicate<String> {
 
 // Adds a custom "sortedEqual" predicate that compares two `Document`s and returns true if they
 // have the same key/value pairs in them
-public func sortedEqual(_ expectedValue: BSONDocument?) -> Predicate<BSONDocument> {
+public func sortedEqual<T: SortedEquatable>(_ expectedValue: T?) -> Predicate<T> {
```

Comment:
In order to extend this matcher to `BSON` (which may contain arrays with documents or documents themselves), I introduced a protocol and made the matcher generic over it. This became more useful as more things became unordered due to the usage of ExtrasJSON instead of Foundation.